### PR TITLE
Introduce VMware.vSphereDSC.Logging module 

### PR DESCRIPTION
**Pull request by SimeonGerginov**
### Added
- VMware.vSphereDSC.Logging module is introduced.
- Added Write-VerboseLog and Write-WarningLog functions to Logging module.

### Changed
- Changed logging mechanism from Write-Verbose to Write-VerboseLog and Write-Warning to Write-WarningLog.

### CODE REVIEW

**Comment by SimeonGerginov** [comment]
Example usage of Write-VerboseLog:

```
 Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
```

is modified to:

```
 Write-VerboseLog -Message "{0} Entering {1}" -Arguments @((Get-Date), (Get-PSCallStack)[0].FunctionName)
```

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+<#
+.SYNOPSIS
+Writes the specified message to the verbose message stream.
+
+.DESCRIPTION
+Writes the specified message to the verbose message stream. If arguments
+are passed to the function, the message is formatted accordingly.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Say that string.Format is used when arguments are specified

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+.PARAMETER Message
+Specifies the message to write to the verbose message stream.
+
+.PARAMETER Arguments
+Specifies the arguments that are needed for formatting the message.
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Arguments for string.Format, message is expected to be a format string that expects given arguments

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 [hunk 3]
```diff
@@ -45,3 +45,36 @@ function Write-VerboseLog {
         Write-Verbose -Message ([string]::Format($Message, $Arguments))
     }
 }
+
+<#
```
Comment (dmilov) [inline Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Same comments as the Write-VerboseLog

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 (hunk 1 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Done.

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 (hunk 2 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Done.

File: Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1 (hunk 3 above)
Comment (SimeonGerginov) [reply Source/VMware.vSphereDSC/VMware.vSphereDSC.Logging.psm1]: Done.

**Pull request merged**